### PR TITLE
Update baseClient.ts

### PR DESCRIPTION
**Pull request by timthedev07**
New method

### CODE REVIEW

File: src/baseClient.ts [hunk 1]
```diff
@@ -84,6 +84,35 @@ export class ClasschartsClient {
       }
     );
   }
+  
+  /**
+   * Returns all of the activities from and to the indicated dates.
+   * An abstraction on top of the getActivity method, it fetches everything in the given interval instead of returning a fragment.
+   */
+  async getFullActivity(from: string, to: string, client: ClasschartsClient) {
```
Comment (jamesatjaminit) [inline src/baseClient.ts:92]:
```suggestion
  async getFullActivity(options:getFullActivityOptions) {
```
I'd prefer it if we could use a object like the other functions (obviously updating the type)

Comment (jamesatjaminit) [inline src/baseClient.ts:92]:
```suggestion
  async getFullActivity(from: string, to: string) {
```
The client shouldn't be needed here, see comment below

File: src/baseClient.ts [hunk 2]
```diff
@@ -84,6 +84,35 @@ export class ClasschartsClient {
...
+    
+    while (true) {
+      const params: GetActivityOptions = {
+        from,
+        to,
+      };
+      if (prevLast) {
+        params.last_id = `${prevLast}`;
+      }
+      const fragment = await client.getActivity(params);
```
Comment (jamesatjaminit) [inline src/baseClient.ts:104]:
```suggestion
      const fragment = await this.getActivity(params);
```
We can use the existing client

**Review by jamesatjaminit — CHANGES_REQUESTED** [review]

**Comment by jamesatjaminit** [comment]
Thanks for this, an issue which has been on my mind for a while. Not sure whether the best place for it is in a function in itself or the existing function but this will do for now.

**Comment by timthedev07** [comment]
Thanks! I originally had a standalone function so that's why the first version didn't fit well into a class method.

**Comment by jamesatjaminit** [comment]
Implemented in https://github.com/Classcharts-API/classcharts-api/commit/eea157566c90f6ac609dfd7df9803f85294cc636

**Pull request closed without merging**